sql/scan.go

```diff
@@ -102,24 +112,68 @@ func (n *scanNode) Err() error {
 	return n.err
 }
 
-func (n *scanNode) init() bool {
-	return n.initScan() && n.initExprs()
-}
+func (n *scanNode) initFrom(p *planner, from parser.TableExprs) error {
+	switch len(from) {
+	case 0:
+		// n.desc remains nil.
+		return nil
 
-// initExprs initializes the render and filter expressions for the
-// scan. Initialization consists of replacing QualifiedName nodes with
-// ParenExpr nodes for which the wrapped expression can be changed for each
-// row.
-func (n *scanNode) initExprs() bool {
-	n.qvals = make(qvalMap)
-	for i := range n.render {
-		n.render[i], n.err = n.extractQVals(n.render[i])
-		if n.err != nil {
-			return false
+	case 1:
```

Comment:
this looks like a lot of code movement that is unrelated to the "real" change in this PR. can it at least get its own commit?